packages/perseus-editor/src/widgets/interactive-graph-editor/interactive-graph-editor.tsx

```diff
@@ -694,6 +751,54 @@ class InteractiveGraphEditor extends React.Component<Props> {
     }
 }
 
+// Merges two graphs that have the same `type`. Properties defined in `b`
+// overwrite properties of the same name in `a`. Throws an exception if the
+// types are different or not recognized.
+function mergeGraphs(
+    a: PerseusGraphType,
+    b: PerseusGraphType,
+): PerseusGraphType {
+    if (a.type !== b.type) {
+        throw new Error(
+            `Cannot merge graphs with different types (${a.type} and ${b.type})`,
+        );
+    }
+    switch (a.type) {
```

Comment:
Do we need this switch statement if there's already a check for `a.type !== b.type` above?

I feel like we should just be able to return `{...a, ...b}` after that? Or is this a case of TypeScript complaining if you do that?

Reply:
Yeah, TypeScript can't figure it out.